packages/main/src/plugin/container-registry.ts

```diff
@@ -2028,26 +2024,59 @@ export class ContainerProviderRegistry {
         `Uploading the build context from ${containerBuildContextDirectory}...Can take a while...\r\n`,
       );
       const tarStream = tar.pack(containerBuildContextDirectory);
-      if (isWindows() && relativeContainerfilePath !== undefined) {
-        relativeContainerfilePath = relativeContainerfilePath.replace(/\\/g, '/');
+      if (isWindows() && options?.containerFile !== undefined) {
+        options.containerFile = options.containerFile.replace(/\\/g, '/');
       }
 
       let streamingPromise: Stream;
       try {
-        streamingPromise = (await matchingContainerProviderApi.buildImage(tarStream, {
+        const buildOptions: ImageBuildOptions = {
           registryconfig,
-          dockerfile: relativeContainerfilePath,
-          t: imageName,
-          platform: platform,
-          abortSignal: abortController?.signal,
-        })) as unknown as Stream;
+          abortSignal: options?.abortController?.signal,
+          dockerfile: options?.containerFile,
+          t: options?.tag,
+          platform: options?.platform,
+          remote: options?.remote,
+          q: options?.q,
+          pull: options?.pull,
+          rm: options?.rm,
+          forcerm: options?.forcerm,
+          memory: options?.memory,
+          memswap: options?.memswap,
+          cpushares: options?.cpushares,
+          cpusetcpus: options?.cpusetcpus,
+          cpuperiod: options?.cpuperiod,
+          cpuquota: options?.cpuquota,
+          shmsize: options?.shmsize,
+          squash: options?.squash,
+          networkmode: options?.networkmode,
+          target: options?.target,
+          outputs: options?.outputs,
+          nocache: options?.nocache,
+        };
+        if (options?.extrahosts) {
+          buildOptions.extrahosts = options.extrahosts;
+        }
+        if (options?.cachefrom) {
+          buildOptions.cachefrom = options.cachefrom;
+        }
+        if (options?.buildargs) {
+          buildOptions.buildargs = options.buildargs;
+        }
+        if (options?.labels) {
+          buildOptions.labels = options.labels;
+        }
```

Comment:
A question by just looking at the code? Why didn't you add them inside the object above like the other props?
Like `extrahosts: options?.extrahosts`

Reply:
Dockerode uses the presence of the property to send an empty string to the engine and this causes an error because a JSON string is expected